frontend_tests/node_tests/compose_ui.js

```diff
@@ -74,6 +74,58 @@ run_test('insert_syntax_and_focus', () => {
     assert.equal($('#compose-textarea').val(), 'xyz :octopus: ');
     assert($("#compose-textarea").is_focused());
 
+    // Tests for block_format option in various cases
+    var textarea = make_textbox('');
+    // Case 1: "|"
+    textarea.caret(0);
+    compose_ui.insert_syntax_and_focus(':sharky:', textarea, true);
```

Comment:
I know this will make things more verbose, but I think we should make the examples for this be actual block syntax, e.g. `~~~[Uploading filename…]()~~~`

~~~

Reply:
Alright, I've replaced `:sharky:` with the string `[Uploading random_file...]()`. One doubt though, since in the new commit, the main changes happen in the `smart_insert` function, I've moved the tests to `smart_insert`.  Should I add for unit tests for `insert_syntax_and_focus` too?